app/models/track.rb

```diff
@@ -42,7 +46,7 @@ def self.for_repo(repo)
   end
 
   def recache_num_exercises!
-    update_column(:num_exercises, exercises.where(status: %i[active beta]).count)
+    update_column(:num_exercises, exercises.published(self).count)
```

Comment:
I've moved the `active`/`beta` logic into a scope on `Exercise`. It was repeated in a couple of places.